classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,460 @@
+     */
+    public function filterByVolumes(array $volumes): self
+    {
+        if (!is_null($volumes) && !is_array($volumes)) {
+            $volumes = [$volumes];
+        }
+        $this->volumes = $volumes;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $numbers
+     *
+     * @return $this
+     */
+    public function filterByNumbers(array $numbers): self
+    {
+        if (!is_null($numbers) && !is_array($numbers)) {
+            $numbers = [$numbers];
+        }
+        $this->numbers = $numbers;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $years
+     *
+     * @return $this
+     */
+    public function filterByYears(array $years): self
+    {
+        if (!is_null($years) && !is_array($years)) {
+            $years = [$years];
+        }
+        $this->years = $years;
+        return $this;
+    }
+
+    /**
+     * Set issue id filter
+     *
+     * @return $this
+     */
+    public function filterByIds(array $issueIds): self
+    {
+        $this->issueIds = $issueIds;
+        return $this;
+    }
+
+    /**
+     * set urlPath filter
+     *
+     * @return $this
+     */
+    public function filterByUrlPath(string $urlPath): self
+    {
+        $this->urlPath = $urlPath;
+        return $this;
+    }
+
+    /**
+     * Set titles filter
+     *
+     * @return $this
+     */
+    public function filterByTitles(array $titles): self
+    {
+        $this->titles = $titles;
+        return $this;
+    }
+
+    /**
+     * Set query search phrase
+     *
+     * @return $this
+     */
+    public function searchPhrase(string $phrase): self
+    {
+        $this->searchPhrase = $phrase;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     *
+     * @return $this
+     */
+    public function limit(int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     *
+     * @return $this
+     */
+    public function offset(int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $this->columns[] = 'i.*';
+        $q = DB::table($this->dao->table, 'i')
+            ->leftJoin('issue_settings as iss', 'i.issue_id', '=', 'iss.issue_id')
+            ->leftJoin('custom_issue_orders as o', 'o.issue_id', '=', 'i.issue_id')
+            // TODO: Is this still necessary (not just for search phrase, but also others)
+            ->groupBy('i.issue_id', $this->orderColumn);
+
+        // Issue titles (exact matches)
+        foreach ($this->titles as $key => $title) {
+            $tableId = 'iss' . $key;
+            $q->join('issue_settings as ' . $tableId, function (JoinClause $join) use ($tableId, $title) {
```

Comment:
This should probably be converted to a `WHERE i.issue_id IN (subquery)` form; otherwise multiple matches here will result in multiple rows, which need to then be resolved using `GROUP BY`.